### DISCUSSION TITLE
http extension has issues with types when using dynamic values

### DISCUSSION THREAD

**Discussion by mclean25** [supabase/supabase#11121] [Questions]
# Bug report

Using the `http` function, anytime I use a dynamic value I get an error saying `function {x}({type}) does not exist`. For example, this code will produce the error: `function urlencode(character varying) does not exist`. Note that from [the http docs](https://github.com/pramsey/pgsql-http#functions) it says that `varchar is indeed the expected type`.

```sql
create or replace function call_status(org_id varchar)
returns int
language sql
as $$
  select status from http_post(
    CONCAT('https://google.com/', urlencode(jsonb_build_object('orgId' , org_id)::varchar))
  )
$$;
```

Also, if you do something like the below where the url is dynamically injected from the params, you get a similar issue of `function http_get(character varying) does not exist`:

```sql
create or replace function call_status(url varchar)
returns int
language sql
as $$
  select status from http_get(
    url
  )
$$;
```

## Additional context

This was all done running Supabase CLI 1.26.7 using the SQL console.

**Comment by GaryAustin1** [comment]
Hopefully someone can help you with your issue, but I'll just say it is very highly unlikely this has anything to do with Supabase and is more a help question than a bug.
Everything you are using is Postgres SQL specific.
Mainly saying this so you don't wait around, and start looking at the broader Postgres help on the web.

**Comment by GaryAustin1** [comment]
Did you happen to install the http extension in the dashboard/database/extensions settings?

**Comment by kiwicopple** [comment]
thanks Gary - indeed this seems more like worth raising on the http extension repo:  https://github.com/pramsey/pgsql-http

I'll convert this to a discussion for now, in case anyone has some tips
